### PR TITLE
General packaging improvements

### PR DESCRIPTION
**Pull request by nike4613**
Finally, sane NuGet package creatin!

This PR incorporates [NuGetizer](https://github.com/devlooped/nugetizer) in place of the stock NuGet package creation for better control.

Notably, it:
- Makes Hive.ReferenceAssembly *actually* contain a reference assembly
- Cleans up some of the standalone packaging work
- Makes the analyzer packages be shaped like standard analyzer packages
- Adds a script which generates all NuGet packages.

This PR also, mostly incidentally, moves the `bin/` and `obj/` folders from in-source to the `artifacts/` folder. This means that all build artifacts of any kind are now contained there. 

This was not worth staying up this late.

### CODE REVIEW

**Review by Caeden117 — APPROVED** [review]
rootin' tootin' package creatin'🤠

LGTM

File: src/Hive/Hive.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFrameworks>net5.0</TargetFrameworks>
```
Comment (sc2ad) [inline src/Hive/Hive.csproj:4]: Is this necessary/what changes between these two?

File: tools/Artifacts.props [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    
+	<PropertyGroup>
+		<HiveArtifactsPath>$(MSBuildThisFileDirectory)..\artifacts\</HiveArtifactsPath>
```
Comment (sc2ad) [inline tools/Artifacts.props:5]: Does this always suffix with `/` or `\`?

**Review by sc2ad — APPROVED** [review]
Aside from some questions, looks good

File: src/Hive/Hive.csproj (hunk 1 above)
Comment (nike4613) [reply src/Hive/Hive.csproj:4]: *Strictly* the change isn't necessary, but it means that during the build invocation `TargetFramework` is set globally, no matter where in evaluation you are, which is rather helpful. It means, among other things, that I can access `TargetFramework` in projects imported before that `PropertyGroup`.

File: tools/Artifacts.props (hunk 2 above)
Comment (nike4613) [reply tools/Artifacts.props:5]: Yes, as is convention in MSBuild. Anything referring to a directory always has a trailing slash.

**Pull request merged**